packages/legend-data-cube/src/stores/core/DataCubeQuerySnapshotBuilderUtils.ts

```diff
@@ -121,3 +166,291 @@ export function _pruneExpandedPaths(
     )
     .sort();
 }
+
+export function _buildFilterSnapshot(
+  vs: V1_ValueSpecification,
+): DataCubeQuerySnapshotFilter {
+  let stack = [vs];
+
+  let filterConditionSnapshot = [];
+  let filterSnapshot = {
+    groupOperator: DataCubeQueryFilterGroupOperator.AND,
+  } as DataCubeQuerySnapshotFilter;
+  if (vs instanceof V1_AppliedFunction) {
+    switch (vs.function) {
+      case DataCubeQueryFilterGroupOperator.OR:
+      case DataCubeQueryFilterGroupOperator.AND:
+        filterSnapshot.groupOperator = vs.function;
+        stack.pop();
+        vs.parameters.forEach((param) => stack.push(param));
+        break;
+    }
+  }
+  // filterSnapshot.groupOperator = DataCubeQueryFilterGroupOperator.AND;
+  while (stack.length > 0) {
+    let toApply = stack.shift();
+    if (toApply instanceof V1_AppliedFunction) {
```

Comment:
see my other comment in `DataCubeQueryFilterOperation`, the logic should be offloaded to each operator, this would simplify the logic here down to just walking the tree of AND/OR.

Reply:
done